components/salsa-macro-rules/src/setup_tracked_struct.rs

```diff
@@ -231,6 +231,16 @@ macro_rules! setup_tracked_struct {
                         $zalsa::None
                     }
                 }
+
+                #[inline]
+                unsafe fn memo_table(
+                    zalsa: &$zalsa::Zalsa,
+                    id: $zalsa::Id,
+                    current_revision: $zalsa::Revision,
+                ) -> $zalsa::MemoTableWithTypes<'_> {
+                    // SAFETY: Guaranteed by caller.
```

Comment:
Can we extend the SAFETY documentation on the `memo_table` definition. It only mentions the constraints on `current_revision` but I don't think this is enough to guareantee the memo's type?